libqtile/backend/x11/core.py

```diff
@@ -744,18 +766,29 @@ def painter(self):
         return self._painter
 
     def simulate_keypress(self, modifiers, key):
-        """Simulates a keypress on the focused window."""
-        # FIXME: This needs to be done with sendevent, once we have that fixed.
+        """Simulates a keypress."""
         modmasks = xcbq.translate_masks(modifiers)
         keysym = xcbq.keysyms.get(key)
-
-        class DummyEv:
-            pass
-
-        d = DummyEv()
-        d.detail = self.conn.keysym_to_keycode(keysym)[0]
-        d.state = modmasks
-        self.handle_KeyPress(d)
+        keysym_code = self.conn.keysym_to_keycode(keysym)[0]
+
+        child = 0
+        if self.qtile.current_window:
+            child = self.qtile.current_window.wid
+
+        ev = KeyPressEvent.synthetic(
+            child=child,
+            detail=keysym_code,
+            event=self._root.wid,
+            event_x=0,
+            event_y=0,
+            root=self._root.wid,
+            root_x=0,
+            root_y=0,
+            same_screen=1,
```

Comment:
Perhaps it would be better to use the real pointer coordinates? Otherwise windows will always receive the events in the top left corner.

Reply:
It is but they're keypresses not mouse presses so does this make a difference?

Reply:
Ah yeah, my mistake. I assume it doesn't make a difference, and perhaps the arguments are there because the signature might match a generic synthetic event signature or something.

Reply:
Well x11 does actually give the pointer x and y on real events when I logged them, so your idea is not wrong

Reply:
Implemented.